lib/minitest/mock.rb

```diff
@@ -222,8 +227,10 @@ def stub name, val_or_callable, *block_args
 
     yield self
   ensure
-    metaclass.send :undef_method, name
-    metaclass.send :alias_method, name, new_name
-    metaclass.send :undef_method, new_name
+    unless metaclass.nil?
```

Comment:
Indeed, I thought about saving the boolean as well. Like this:

```ruby
restubbed = respond_to? new_name
if restubbed
  # Raise
end
```

And then use this variable as well in the `ensure` block.
It is not much cleaner, but it is explicit and more future proof.

What do you think?